lib/src/rules/prefer_contains.dart

```diff
@@ -79,37 +78,22 @@ class _Visitor extends SimpleAstVisitor<void> {
   _Visitor(this.rule, this.context);
 
   @override
-  void visitSimpleIdentifier(SimpleIdentifier node) {
-    // Should be "indexOf".
-    var propertyElement = node.staticElement;
-    if (propertyElement?.name != 'indexOf') {
+  void visitMethodInvocation(MethodInvocation node) {
+    if (node.methodName.name != 'indexOf') {
       return;
     }
 
-    MethodInvocation indexOfAccess;
-    InterfaceType type;
-
-    var parent = node.parent;
-    if (parent is MethodInvocation && node == parent.methodName) {
-      indexOfAccess = parent;
-      var parentType = parent.target?.staticType;
-      if (parentType is! InterfaceType) {
-        return;
-      }
-      type = parentType;
-    } else {
-      return;
-    }
-
-    if (!DartTypeUtilities.implementsAnyInterface(
-        type, <InterfaceTypeDefinition>[
-      InterfaceTypeDefinition('Iterable', 'dart.core'),
-      InterfaceTypeDefinition('String', 'dart.core'),
-    ])) {
+    var parentType = node.target?.staticType;
+    if (parentType == null ||
+        !DartTypeUtilities.implementsAnyInterface(
+            parentType, <InterfaceTypeDefinition>[
```

Comment:
nit: Note that you probably don't need the type argument list in front of the list literal. It ought to be inferred.

Reply:
Good catch.